website/profile/views.py

```diff
@@ -701,6 +701,7 @@ def unserialize_social(auth, **kwargs):
 
     try:
         user.save()
+        user.check_spam(saved_fields=None, request_headers=request.headers)
```

Comment:
We are outside the Django request context here.

Reply:
Should we put a `.check_spam` call in `OSFUser.save` instead? The model has `SPAM_USER_PROFILE_FIELDS` already, and the `save` function checks `dirty_fields`. Following same pattern as `SEARCH_UPDATE_FIELDS` (and `_get_spam_content`), it might look something like
```
### imports
# get_current_request already being imported

from osf.utils.requests import get_current_request, get_headers_from_request

### def save

# [...]
if set(self.SPAM_USER_PROFILE_FIELDS.keys()).intersection(dirty_fields):
    request = get_current_request()
    headers = get_headers_from_request(request)
    self.check_spam(dirty_fields, request_headers=headers)
```

Reply:
Yes, this would seem to be the obvious thing to do, but `api/users/serializers.py` ln 255, establishes the user spam check in the update method of the `UserSerializer`.  If you look at those other views you can see on `website/profile/views.py` ln 754, it's an established pattern for v1, if a bad one. 

I will test that code snippet and see if it works with the tests. Hopefully that doesn't create too many double checks on serializer updates or v1 routes. Just wanted to give some context for the original PR.